namesys/interface.go

```diff
@@ -47,6 +47,14 @@ const (
 	// trust resolution to eventually complete and can't put an upper
 	// limit on how many steps it will take.
 	UnlimitedDepth = 0
+
+	// ImmutableTTL is the time-to-live value to be reported for immutable
+	// paths.
```

Comment:
Fair enough. It would be easy to create a data type like `data TTLDuration = Finite Duration | Infinity` in languages with custom sum types, and the ordering of the constructors would make `min`/`max` do the right thing, too. I wonder what the cleanest thing to do is in Go?

* Floats? (Eww)
* `interface {}` and dynamic typing? (Eww, too)
* A pointer to a Duration with nil signifying infinity?
* Something less ugly that I didn’t think of? (I am still new to Go.)